src/serac/infrastructure/profiling.hpp

```diff
@@ -195,11 +197,14 @@ void startCaliperRegion(const char* name);
  */
 void endCaliperRegion(const char* name);
 
+/// @brief Provides the member typedef type that names T.
```

Comment:
Might be useful to link to https://en.cppreference.com/w/cpp/types/type_identity which also has a note on why it's useful